internal/impl/aws/input_kinesis.go

```diff
@@ -671,6 +692,35 @@ func (k *kinesisReader) runExplicitShards() {
 	}
 }
 
+func (k *kinesisReader) maxShardRunBalancedConsumer(wg *sync.WaitGroup, streamID string, unclaimedShards map[string]string, numShardAlreadyClaimed int) (int, error) {
+	numShardClaimed := 0
+	maxShardNumber, maxShardExists := k.streamMaxShards[streamID]
+	if !maxShardExists || numShardAlreadyClaimed < maxShardNumber {
+		// Remaining number of shards that can be claimed
+		for shardID, clientID := range unclaimedShards {
+			if maxShardExists && numShardAlreadyClaimed + numShardClaimed >= maxShardNumber {
+				break
+			}
+			sequence, err := k.checkpointer.Claim(k.ctx, streamID, shardID, clientID)
+			if err != nil {
+				if k.ctx.Err() != nil {
+					return 0, err
+				}
+				if !errors.Is(err, ErrLeaseNotAcquired) {
+					k.log.Errorf("Failed to claim unclaimed shard '%v': %v\n", shardID, err)
+				}
+				continue
+			}
+			wg.Add(1)
+			numShardClaimed++
+			if err = k.runConsumer(wg, streamID, shardID, sequence); err != nil {
+				k.log.Errorf("Failed to start consumer: %v\n", err)
```

Comment:
```suggestion
				numShardClaimed--
				k.log.Errorf("Failed to start consumer: %v\n", err)
```

Reply:
Why would this be a decrement?

Reply:
So if the consuming fails it means we could not claim the shard, right? If you look at the `runConsumer` implementation it's also calling the `wg.Done` on error

Reply:
Oh right! its in the `if` statement. Gotcha 👍